projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -309,23 +332,28 @@ int32_t gpio_get(struct gpio_desc **desc,
 		 uint8_t gpio_number)
 {
 	gpio_desc *descriptor;
+	extra_gpio_desc *extra_descriptor;
 	int32_t ret;
 
 	descriptor = (struct gpio_desc *) malloc(sizeof(*descriptor));
-	if (!descriptor)
+	extra_descriptor = (struct extra_gpio_desc *) calloc(1,
+			   sizeof(*extra_descriptor));
```

Comment:
i would use the structure used to define extra_descriptor, not referencing the actual pointer.